examples/transactions.c

```diff
@@ -184,135 +154,137 @@ create_transactional_producer (const rd_kafka_topic_partition_t *rktpar) {
         if (error)
                 fatal_error("init_transactions()", error);
 
-
-        /* Begin a new transaction */
-        error = rd_kafka_begin_transaction(rk);
-        if (error)
-                fatal_error("begin_transaction()", error);
-
         return rk;
 }
 
 
 /**
- * @brief Abort the current transaction and destroy the producer.
+ * @brief Rewind consumer's consume position to the last committed offsets
+ *        for the current assignment.
  */
-static void destroy_transactional_producer (rd_kafka_t *rk) {
+static void rewind_consumer (rd_kafka_t *consumer) {
+        rd_kafka_topic_partition_list_t *offsets;
+        rd_kafka_resp_err_t err;
         rd_kafka_error_t *error;
+        int i;
 
-        fprintf(stdout, "%s: aborting transaction and terminating producer\n",
-                rd_kafka_name(rk));
+        /* Get committed offsets for the current assignment, if there
+         * is a current assignment. */
+        err = rd_kafka_assignment(consumer, &offsets);
+        if (err) {
+                fprintf(stderr, "No current assignment to rewind: %s\n",
+                        rd_kafka_err2str(err));
+                return;
+        }
 
-        /* Abort the current transaction, ignore any errors
-         * since we're terminating the producer anyway. */
-        error = rd_kafka_abort_transaction(rk, -1);
-        if (error) {
-                fprintf(stderr,
-                        "WARNING: Ignoring abort_transaction() error since "
-                        "producer is being destroyed: %s\n",
-                        rd_kafka_error_string(error));
-                rd_kafka_error_destroy(error);
+        if (offsets->cnt == 0) {
+                fprintf(stderr, "No current assignment to rewind\n");
+                rd_kafka_topic_partition_list_destroy(offsets);
+                return;
         }
 
-        rd_kafka_destroy(rk);
-}
+        /* Note: Timeout must be lower than max.poll.interval.ms */
+        err = rd_kafka_committed(consumer, offsets, 10*1000);
+        if (err)
+                fatal("Failed to acquire committed offsets: %s",
+                      rd_kafka_err2str(err));
+
+        /* Seek to committed offset, or start of partition if no
+         * committed offset is available. */
+        for (i = 0 ; i < offsets->cnt ; i++) {
+                /* No committed offset, start from beginning */
+                if (offsets->elems[0].offset < 0)
+                        offsets->elems[0].offset =
+                                RD_KAFKA_OFFSET_BEGINNING;
+        }
 
+        /* Perform seek */
+        error = rd_kafka_seek_partitions(consumer, offsets, -1);
```

Comment:
what if this blocks longer than `max.poll.interval.ms`? should this also have a positive timeout?

Reply:
what if neither `seek_partitions` or `committed` blocks for > 10s, but combined they do?

i guess you'll get a max poll interval error in the consume loop. i notice there you just log it though, whereas timeouts above are considered fatal.

Reply:
We could provide a `rd_kafka_get_remaing_time_until_i_need_to_poll_again(consumer, desired_max_time)` which can be used as a timeout, but we can't really change the timeout behaviour of existing calls.

But I doubt it matters in practice, if any of these (seek, commit, max.poll, ..) fail the application needs to perform some abortive and resetting action anyway, so I'd say it is probably simpler to assume seek, commit, etc, with infinite timeout will finish in time, and if they don't the max poll will kick in and trigger a rebalance - which the application does not necessarily need to handle, things will recover anyway.